packages/drawer/src/index.tsx

```diff
@@ -15,11 +16,32 @@ export interface IDrawerProps extends JSX.HTMLAttributes {
   modal?: boolean;
 }
 
-export interface IDrawerState {}
+enum SLIDER_STATES {
+  REST,
+  SWIPE_START,
+  SWIPING,
+  SWIPE_TO_BE_COMPLETED
+}
+
+export interface IDrawerState {
+  sliderState: SLIDER_STATES;
+  touchPositionX: number;
+  touchPositionY: number;
+  swipeDistance: number;
+}
+
+const CLOSED_STATE = {
+  sliderState: SLIDER_STATES.REST,
+  swipeDistance: -256,
```

Comment:
Is this okay to be hardcoded?

Reply:
we dont really have a good way of detecting this... so our best bet are our visual tests

Reply:
We will see

Reply:
yeah i really dont have a good way of calculating this right now